### DISCUSSION TITLE
QMK (and Via and Vial) Config for 3x5

### DISCUSSION THREAD

**Discussion by zvakanaka** [juhakaup/keyboards#38] [Show and tell]
I'm new to QMK/keyboards. I'm building a 3x5 and was able to get the left side working after some changes to crkbd (have not wired the right side yet).

Compiling with QMK cli required disabling RGB and OLED features in `rules.mk` (I think the firmware was waiting for OLED i2c before allowing the keys to work):
```
RGB_MATRIX_ENABLE = no
RGBLIGHT_ENABLE = no
OLED_ENABLE = no
```

At first it was mirrored/backwards/reversed (typing "qwert" on this left side produced "trewq" and the same goes for the other rows too: "asdfg" was "gfdsa" and "zxcvb" was "bvcxz"). Changing `info.json` `matrix_pins.cols` to have the last 5 columns reversed fixed it.

<details>
<summary>Compile Details</summary>

Here's the compile:
```
qmk config compile.keyboard=crkbd/rev1 compile.keymap=default

qmk compile -e FORCE_LAYOUT=LAYOUT_split_3x5_3
```

This is only needed for Linux to flash to a Pro Micro:
```
avrdude -p m32U4 -P /dev/ttyACM0 -c avr109 -U flash:w:crkbd_rev1_default_LAYOUT_split_
3x5_3.hex
```
</details>

Is there a QMK firmware that would immediately work with this keyboard for the 3x5 version? I am considering using dactyl minidox QMK for a future build, since that looks very similar.

**Comment by zvakanaka** [comment]
I finished the build. One thing that got me was I assumed the right side's columns were mirrored, but they are in the same order as the left side. A wiring diagram for the right half would be awesome for those building this.

Here's a vial-qmk fork with a pin matrix to match your readme's wiring diagram, RGB lighting disabled, and the "unlock" keys set to the top-left on the 3x5: https://github.com/vial-kb/vial-qmk/compare/vial...zvakanaka:vial-qmk:vial?expand=1

Thank you for the excellent readme!

**Comment by juhakaup** [comment]
I converted this to discussion, so that it's easier to find.

Thanks for the fork, I haven't looked into vial that much but I'll check it out later.

I've wired the columns as mirrored in my builds. Are they not mirrored in some version of corne?
I made a more 'complete' wiring diagram for the 4x6 version [here](https://github.com/juhakaup/keyboards/blob/main/Cygnus%20v1.0/4x6_build_notes.md).

**Reply by zvakanaka** [reply]
I looked at my wiring again and realized I made a mistake and reversed the columns! I'm sorry for not looking into this further earlier.

The default [qmk-vial crkbd keymap](https://github.com/vial-kb/vial-qmk/tree/vial/keyboards/crkbd/keymaps/vial) may have worked without modification, had I not wired the columns up mirrored.
